### DISCUSSION TITLE
Repology package update isn't noticed

### DISCUSSION THREAD

**Discussion by jbergstroem** [renovatebot/renovate#20313] [Request Help] [answered]
TL;DR: If you end up here because of google or other search: the reason repology updates may not work for you is because you should be using the versioning scheme `loose`. This obviously varies per package/repository which in my case was packages in Alpine Linux.

---

### Describe the bug

Given the following [config snippet](https://github.com/jbergstroem/mariadb-alpine/blob/main/.github/renovate.json#L39-L44):

```json
{
  "regexManagers": [
    {
      "fileMatch": ["^Dockerfile$"],
      "matchStrings": ["ARG APK_VERSION=\"(?<currentValue>.*?)\"\\n"],
      "datasourceTemplate": "repology",
      "depNameTemplate": "alpine_3_17/mariadb"
    },
  ]
}
```

..and this entry in [my Dockerfile](https://github.com/jbergstroem/mariadb-alpine/blob/main/Dockerfile#L6):
```ini
ARG APK_VERSION="10.6.11-r0"
```

..and this snippet from the [api response](https://repology.org/api/v1/project/mariadb):
```json
{
   "repo":"alpine_3_17",
   "subrepo":"main",
   "srcname":"mariadb",
   "binname":"mariadb-dev",
   "visiblename":"mariadb-dev",
   "version":"10.6.12",
   "maintainers":[
      "ncopa@alpinelinux.org"
   ],
   "licenses":[
      "GPL-2.0-or-later"
   ],
   "summary":"A fast SQL database server (development files)",
   "status":"outdated",
   "origversion":"10.6.12-r0"
}
```

I would assume that renovate should flag the package as outdated - but the debug logs has an empty `updates` array (renovatebot log id: `1001097382`):
```json
{
   "regex": [
      {
        "datasourceTemplate": "repology",
        "depNameTemplate": "alpine_3_17/mariadb",
        "deps": [
          {
            "currentValue": "10.6.11-r0",
            "currentVersion": "10.6.11-r0",
            "datasource": "repology",
            "depIndex": 0,
            "depName": "alpine_3_17/mariadb",
            "fixedVersion": "10.6.11-r0",
            "registryUrl": "https://repology.org",
            "replaceString": "ARG APK_VERSION=\"10.6.11-r0\"\n",
            "versioning": "semver",
            "warnings": [],
            "updates": []
          }
        ],
        "matchStrings": [
          "ARG APK_VERSION=\"(?<currentValue>.*?)\"\\n"
        ],
        "packageFile": "Dockerfile"
      }
    ]
}
```

### Relevant debug logs

```json
{
   "regex": [
      {
        "datasourceTemplate": "repology",
        "depNameTemplate": "alpine_3_17/mariadb",
        "deps": [
          {
            "currentValue": "10.6.11-r0",
            "currentVersion": "10.6.11-r0",
            "datasource": "repology",
            "depIndex": 0,
            "depName": "alpine_3_17/mariadb",
            "fixedVersion": "10.6.11-r0",
            "registryUrl": "https://repology.org",
            "replaceString": "ARG APK_VERSION=\"10.6.11-r0\"\n",
            "versioning": "semver",
            "warnings": [],
            "updates": []
          }
        ],
        "matchStrings": [
          "ARG APK_VERSION=\"(?<currentValue>.*?)\"\\n"
        ],
        "packageFile": "Dockerfile"
      }
    ]
}
```

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by jbergstroem** [comment, marked as answer]
Will test if this is related to versioning scheme.

Edit: yes; you need to use `loose` versioning most likely due to `-r0`:
```json
"updates": [
  {
    "bucket": "non-major",
    "newVersion": "10.6.12-r0",
    "newValue": "10.6.12-r0",
    "newMajor": 10,
    "newMinor": 6,
    "updateType": "patch",
    "branchName": "renovate/all-minor-patch"
  }
],
```
